public/apps/resource-sharing/resource-sharing-panel.tsx

```diff
@@ -0,0 +1,850 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { useEffect, useMemo, useState } from 'react';
+import {
+  EuiEmptyPrompt,
+  EuiBasicTable,
+  EuiBadge,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiComboBox,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+  EuiToolTip,
+  EuiSuperSelect,
+  EuiSuperSelectOption,
+} from '@elastic/eui';
+
+import { titleCase } from 'text-case';
```

Comment:
Was there an issue with `lodash` here? I can see other usages of lodash in the repo so would prefer to use the existing dependency.